src/core/grid_based_algorithms/lbgpu.hpp

```diff
@@ -238,9 +240,16 @@ uint64_t lb_fluid_get_rng_state_gpu();
 void lb_fluid_set_rng_state_gpu(uint64_t counter);
 uint64_t lb_coupling_get_rng_state_gpu();
 void lb_coupling_set_rng_state_gpu(uint64_t counter);
+
+/** Calculate the node index from its coordinates */
+inline unsigned int calculate_node_index(LB_parameters_gpu const &lbpar,
```

Comment:
```Utils::ravel_index(coord, lbpar.dim)``` could probably used with slight modifications

Reply:
what is the difference between ```Utils::get_linear_index``` and ```Utils::ravel_index```? If they are doing the same, we should drop one of them.

Reply:
`Utils::get_linear_index` is optimized for 3D grids (2 multiplications and 2 additions). `Utils::ravel_index` is only used in the `ReactionEnsemble` and `Histogram`, where we don't always work with 3D grids, and uses a `for` loop.